catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/operation/impl/QueryResponseImpl.java

```diff
@@ -302,7 +302,7 @@ private Result handleTake() {
         Result result = null;
         try {
             result = queue.take();
-            if (result == poisonPillResult) {
+            if (result == POISON_PILL_RESULT) {
```

Comment:
Actually, looking at the code now. Am I crazy or does this result in a memory leak for every single query response that goes through the system. Every response has a pointer back to the same poisonPillResult, which is a static (will never be GC'd). So even after the response itself goes out of scope, it will never be cleaned up because one of its results is a static. Right, wrong, crazy?